index.js

```diff
@@ -45,6 +45,10 @@ function unregisterAll(win) {
 }
 
 function register(win, accelerator, callback) {
+	win.on('close', () => {
+		unregisterAllShortcuts(win);
+	});
+
 	if (arguments.length === 2 && typeof win === 'string') {
```

Comment:
`register` could be called multiple time on the same window, to register multiple shortcuts.
In this case, you'll register multiple handlers and call `unregisterAllShortcuts ` multiple time with no reason.
You could move the handler registration below, just below

```js
windowsWithShortcuts.set(win, [{
	accelerator,
	callback
}]);
``` 

This way we register only a handler for every window...